### ISSUE TITLE
Bug: [no-use-before-define] falsely reports on fields of import statement

### ISSUE THREAD

**Issue by LukeNotable** [typescript-eslint/typescript-eslint#10087]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=4.7.4&fileType=.tsx&code=PTAEAEBcGcFoEsDmA7A9gJwKYCh4FsAHDSUAb1AEFQBfUAM3VT1AHIA6YAQzYCtoWA3NlyFioAJqgAvJTYANNuKH4i6EnOmyFALSHZMAD1UlycgDQSaAoA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tmUVoCNEAMw7Q%2BAE2HNE6MAG1IiaNFGQANFE5DYTMpAC64MAF8QxoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
import { A } from './a.js';

import Y = A.X.Y;  // Error: 'X' was used before it was defined.
import X = A.X.Z;

export { X, Y };
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-use-before-define": ["error", "nofunc"]
  }
}
```

### Expected Result

No error.

### Actual Result

The `X` within `A.X.Z` is falsely flagged as a use-before-define of the following `X`.

**Comment by kirkwaiblinger** [comment]
Thanks for reporting! 

I was able to find a simpler repo that doesn't have any TS errors:

[Playground](https://typescript-eslint.io/play/#ts=4.7.4&fileType=.ts&code=HYQwtgpgzgDiDGEAEBBAdADTQTSQbwF8AoIgSzBgHsAnAFyVwF5VMcBuMim%2BjJZ9DGyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tmUVoCNEAMw7Q%2BAE2HNE6MAG1IAdwCG0JpAA0UTkNhMykALrgwAXxCmgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

```ts
namespace A.X.Y {}

import Y = A.X.Y;
import X = A.X;
```
